miner/miner.go

```diff
@@ -98,20 +99,25 @@ func (miner *Miner) update() {
 			case downloader.StartEvent:
 				wasMining := miner.Mining()
 				miner.worker.stop()
+				canStart = false
 				if wasMining {
 					// Resume mining after sync was finished
 					shouldStart = true
 					log.Info("Mining aborted due to sync")
 				}
 			case downloader.DoneEvent, downloader.FailedEvent:
+				canStart = true
 				if shouldStart {
 					miner.SetEtherbase(miner.coinbase)
 					miner.worker.start()
 				}
 			}
 		case addr := <-miner.startCh:
-			miner.SetEtherbase(addr)
-			miner.worker.start()
+			if canStart {
+				miner.SetEtherbase(addr)
+				miner.worker.start()
+			}
+			shouldStart = true
 		case <-miner.stopCh:
 			miner.worker.stop()
```

Comment:
Here, you need to set `shouldStart` to false, no?

Reply:
yes you're right!